mmeval/metrics/coco_panoptic.py

```diff
@@ -0,0 +1,656 @@
+
+        self.nproc = nproc
+        self.seg_prefix = seg_prefix
+        self.classwise = classwise
+
+        # if ann_file is not specified,
+        # initialize coco api with the converted dataset
+        self._coco_api: Optional[PanopticAPI]  # type: ignore
+
+        if ann_file is not None:
+            with get_local_path(
+                    filepath=ann_file,
+                    backend_args=backend_args) as local_path:
+                self._coco_api = PanopticAPI(annotation_file=local_path)
+            self.categories = self._coco_api.cats
+        else:
+            self._coco_api = None
+            self.categories = None
+
+        self.backend_args = backend_args
+        self.keep_results = keep_results
+        self.direct_compute = direct_compute
+        # TODO: update after logger is supported in BaseMetric
+        self.logger = default_logger if logger is None else logger
+
+        # necessary arguments, which can be obtained from self.dataset_meta.
+        # Note that they cannot be force set during initialization, because
+        # self.dataset_meta may be updated after build Metric.
+        self.label2cat = None
+        self.classes = None
+        self.thing_classes = None
+
+    def add(self, predictions: Sequence[Dict], groundtruths: Sequence[Dict]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add the intermediate results to `self._results`.
+
+        Args:
+            predictions (Sequence[dict]): A sequence of dict. Each dict
+                representing a detection result for an image, with the
+                following keys:
+
+                - image_id (int): Image id.
+                - sem_seg (numpy.ndarray): The prediction of semantic
+                  segmentation.
+                - segm_file (str): Segmentation file name.
+
+            groundtruths (Sequence[dict]): A sequence of dict. If load from
+                `ann_file`, the dict inside can be empty. Else, each dict
+                represents a groundtruths for an image, with the following
+                keys:
+
+                - image_id (int): Image id.
+                - segm_file (str): Segmentation file name.
+                - width (int): The width of the image.
+                - height (int): The height of the image.
+                - segments_info (list, optional): groundtruth information
+                  list. Is necessary if self._coco_api is None.
+        """
+        for prediction, groundtruth in zip(predictions, groundtruths):
+            assert isinstance(prediction, dict), 'The prediciton should be ' \
+                f'a sequence of dict, but got a sequence of {type(prediction)}.'  # noqa: E501
+            assert isinstance(groundtruth, dict), 'The label should be ' \
+                f'a sequence of dict, but got a sequence of {type(groundtruth)}.'  # noqa: E501
+            prediction = self._process_prediction(prediction)
+
+            if self.direct_compute:
```

Comment:
Why not keep the same logic with mmdet?

Reply:
Before refactoring, this flag is according to `out_prefix`, if `out_prefix` is None, it will direct compute, otherwise, it will compute results after all images have been processed. But `direct compute` is a bit faster than the second way. Hence, I think it is necessary to give a flag to speed up.